core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -1321,7 +1321,16 @@ private void internalPerformTranslogRecovery(boolean skipTranslogRecovery, boole
 
         assert indexExists == false || assertMaxUnsafeAutoIdInCommit();
 
+
         final EngineConfig config = newEngineConfig(openMode);
+
+        if (openMode == EngineConfig.OpenMode.OPEN_INDEX_AND_TRANSLOG) {
```

Comment:
can you add a comment why it's important to do it before we  open the engine?

Reply:
sure, added in ea23a0d